src/tests/krate.rs

```diff
@@ -1262,7 +1262,7 @@ fn download() {
 
 #[test]
 fn download_nonexistent_version_of_existing_crate_404s() {
-    let (app, anon, user) = TestApp::init().with_user();
+    let (app, anon, user) = TestApp::with_proxy().with_user();
```

Comment:
This works fine for me locally with just `init()`, as nothing in this test connects to S3 or Github.  Otherwise, this PR looks great to me.

Reply:
Using `init` here is caring too much about the internals of the endpoint. This test shouldn't care whether we 404 before talking to S3 or not, just that something 404s at some point. While getting the 404 to come from S3 will certainly require some more changes (adding a recording), it shouldn't panic if we get to the uploader at all.